packages/ag-charts-enterprise/src/features/annotations/annotations.ts

```diff
@@ -186,7 +259,90 @@ export class Annotations extends _ModuleSupport.BaseModuleInstance implements _M
         );
     }
 
-    private onClick(_event: _ModuleSupport.PointerInteractionEvent<'click' | 'drag-start'>) {
+    private onHoverEditing(event: _ModuleSupport.PointerInteractionEvent<'hover'>) {
+        const {
+            annotationData,
+            annotations,
+            editingStep,
+            editingType,
+            ctx: { cursorManager, updateService },
+        } = this;
+
+        if (!annotationData || !editingType || !editingStep) return;
+
+        const offset = {
+            x: event.offsetX - (this.seriesRect?.x ?? 0),
+            y: event.offsetY - (this.seriesRect?.y ?? 0),
+        };
+        const point = this.invertPoint(offset);
+
+        const valid = this.validateDatumPoint(point);
+        cursorManager.updateCursor('annotations', valid ? undefined : 'not-allowed');
+
+        if (!valid || editingStep === EditingStep.Start) return;
+
+        const datum = annotationData.at(-1);
+        this.active = annotationData.length - 1;
+        const node = annotations.nodes()[this.active];
+
+        if (!datum || !node) return;
+
+        node.toggleActive(true);
+
+        const editingStepFn = this.editingStepFns[editingType]?.[editingStep];
+        if (editingStepFn) {
+            editingStepFn(datum, node, point, offset);
+        }
+
+        updateService.update();
+    }
+
+    private editingStepFns: Partial<Record<AnnotationType, Partial<Record<EditingStep, EditingStepFn>>>> = {
```

Comment:
I'm not massively happy with this, but using a switch statement with `case AnnotationType.Line + EditingStep.End` resulted in function complexity failure. I may refactor this more when we add more annotation types and have a better feel for it.